zerver/tests/test_narrow.py

```diff
@@ -420,9 +421,24 @@ def get_and_check_messages(self,
                           "avatar_url", "recipient_id", "sender_full_name",
                           "sender_short_name", "timestamp", "reactions"):
                 self.assertIn(field, message)
-
         return result
 
+    def message_visibility_test(self, narrow: List[Dict[str, str]],
+                                message_ids: List[int], pivot_index: int) -> None:
+        result = self.get_and_check_messages(dict(narrow=ujson.dumps(narrow),
```

Comment:
The overall approach looks good here.

I would consider not using `get_and_check_messages` here, since we don't really need to check anything about the messages other than the ids, so it's really only two or three lines of code to just call `self.client_get('/messages', ...)` directly and then get the json bit.